test/e2e/catalog_e2e_test.go

```diff
@@ -750,6 +753,119 @@ var _ = Describe("Starting CatalogSource e2e tests", Label("CatalogSource"), fun
 		Expect(registryPods.Items).To(HaveLen(1), "unexpected number of replacement registry pods found")
 	})
 
+	It("delete registry network policy triggers recreation", func() {
```

Comment:
do we need an e2e? could a unit test cover this case?

Reply:
Maybe? I didn't look too closely. The main bit of coverage I wanted here was "is there an NP event that the catalog-operator notices and causes there to be a sync of the relevant CatSrc"

It seemed like there were other e2e's for similar catsrc reconcile behaviors, so I followed the pattern.

Reply:
I'm just worried about adding additional e2es XD but I'd also say that if the effort to make it a unit test is too big, it's not worth the effort...let's proceed as it is